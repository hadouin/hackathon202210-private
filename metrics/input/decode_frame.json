{
	"1": {
		"type": "decode_frame",
		"arguments": {"frame": "3"}
	},
	"2": {
		"type": "decode_frame",
		"arguments": {"frame": "79CC29467860171400E800139A0F57543230313835313730303030303100000000C8FFF600820000841C00000000000000000000003B240901000000000A520F0000000000000000"
        }
	},"3": {
		"type": "decode_frame",
		"arguments": {"frame": "79CC28467860171400E800139A0F57543230313835313730303030303100000000C8FFF600820000841C00000000000000000000003B240901000000000A520F0000000000000000"
        }
	},
    "4": {
		"type": "decode_frame",
		"arguments": {"frame": "69CC29467860171400E800139A0F57543230313835313730303030303100000000C8FFF600820000841C00000000000000000000003B240901000000000A520F0000000000000000"}
	},
    "5": {
		"type": "decode_frame",
		"arguments": {"frame": "7923FE2F053B31F125B7A003001862012D022C022A022D022D022E022E022D022E022F0230020000841C00000000000000000000003B240901000000000A520F0000000000000000"}
	}


}
