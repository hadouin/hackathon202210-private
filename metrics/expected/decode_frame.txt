1 Invalid frame
2 {
    "FOS": "CC29",
    "FFC": "4678",
    "MetrologicalSerialNumber": "60171400E800139A",
    "CustomerSerialNumber": "0F575432303138353137303030303031",
    "Backflow": {
        "value": 0,
        "unit": "m3"
    },
    "SignalStrenght": {
        "value": -56,
        "unit": "dbm"
    },
    "SignalQuality": {
        "value": -10,
        "unit": "dbm"
    },
    "Alarms": "00820000",
    "Battery Remaining": {
        "value": 7300,
        "unit": "days"
    },
    "Operating Time": {
        "value": 0,
        "unit": "days"
    },
    "Customer Location": "0000000000000000",
    "Operator Specific Data": "00",
    "Date_Time": {
        "value": "2000-01-09T04:59:00Z",
        "storage": 1
    },
    "BaseIndex": {
        "value": 0,
        "unit": "m3",
        "storage": 1
    },
    "CompactProfile": [
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T04:44:00Z"
        },
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T04:29:00Z"
        },
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T04:14:00Z"
        },
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T03:59:00Z"
        }
    ]
}
3 Invalid FOS
4 Frame doesn't start with 79
5 {
    "FOS": "23FE",
    "FFC": "2F05",
    "MetrologicalSerialNumber": "3B31F125B7A00300",
    "CustomerSerialNumber": "1862012D022C022A022D022D022E022E",
    "Backflow": {
        "value": 771894530,
        "unit": "m3"
    },
    "SignalStrenght": {
        "value": 12034,
        "unit": "dbm"
    },
    "SignalQuality": {
        "value": 2,
        "unit": "dbm"
    },
    "Alarms": "30020000",
    "Battery Remaining": {
        "value": 7300,
        "unit": "days"
    },
    "Operating Time": {
        "value": 0,
        "unit": "days"
    },
    "Customer Location": "0000000000000000",
    "Operator Specific Data": "00",
    "Date_Time": {
        "value": "2000-01-09T04:59:00Z",
        "storage": 1
    },
    "BaseIndex": {
        "value": 0,
        "unit": "m3",
        "storage": 1
    },
    "CompactProfile": [
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T04:44:00Z"
        },
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T04:29:00Z"
        },
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T04:14:00Z"
        },
        {
            "value": 0,
            "unit": "m3",
            "storage": 1,
            "date": "2000-01-09T03:59:00Z"
        }
    ]
}
